### PR TITLE
add support for bucket level request count per API

### PR DESCRIPTION
**Pull request by harshavardhana**
## Description
add support for bucket-level request count per API

## Motivation and Context
New metrics added to calculate API request count
per bucket, per API.  Captures errors, including
4xx, 5xx HTTP status codes separately.

## How to test this PR?
Create bucket and perform operations such as list, upload, download
you should see new metrics appear under your `bucket` as the label.

## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Optimization (provides speedup with no functional changes)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
- [ ] Fixes a regression (If yes, please add `commit-id` or `PR #` here)
- [ ] Unit tests added/updated
- [ ] Internal documentation updated
- [ ] Create a documentation update request [here](https://github.com/minio/docs/issues/new?label=doc-change,title=Doc+Updated+Needed+For+PR+github.com%2fminio%2fminio%2fpull%2fNNNNN)

### CODE REVIEW

File: cmd/metrics-v2.go [hunk 1]
```diff
@@ -2072,6 +2132,55 @@ func getBucketUsageMetrics() *MetricsGroup {
...
+				metrics = append(metrics, Metric{
+					Description:    getBucketS3RequestsCanceledMD(),
+					Value:          float64(v),
+					VariableLabels: map[string]string{"bucket": bucket, "api": k},
+				})
+			}
+
+			for k, v := range httpStats.totalS3Errors.Load() {
+				metrics = append(metrics, Metric{
+					Description:    getBucketS3RequestsErrorsMD(),
```
Comment (vadmeste) [inline cmd/metrics-v2.go]: I think this metric is just confusing for the users since it counts 4xx. I don't think it is needed

**Review by vadmeste — APPROVED** [review]
LGTM with one comment

**Review by kanagarajkm — APPROVED** [review]

File: cmd/metrics-v2.go (hunk 1 above)
Comment (harshavardhana) [reply cmd/metrics-v2.go]: Okay can remove it

File: cmd/metrics-v2.go [hunk 2]
```diff
@@ -1309,6 +1309,56 @@ func getExpiryPendingTasksMD() MetricDescription {
...
+		Subsystem: requestsSubsystem,
+		Name:      "5xx_" + errorsTotal,
+		Help:      "Total number of S3 requests with (5xx) errors on a bucket",
+		Type:      counterMetric,
+	}
+}
+
+func getBucketS3RequestsCanceledMD() MetricDescription {
+	return MetricDescription{
+		Namespace: s3MetricNamespace,
```
Comment (poornas) [inline cmd/metrics-v2.go]:
```suggestion
		Namespace: bucketMetricNamespace,
```

**Review by poornas — APPROVED** [review]

**Pull request merged**